client/app.jsx

```diff
@@ -1,17 +1,42 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
-import Navigation from './components/navigation.jsx';
-import Header from './components/header.jsx'
-
+import parseRoute from './lib/parse-route.js';
+import Home from './pages/home.jsx';
+import Search from './pages/search.jsx';
+import Header from './components/header.jsx';
 
 class App extends React.Component {
-  render(){
+  constructor(props) {
+    super(props);
+    this.state = {
+      route: parseRoute(window.location.hash)
+    };
+    this.renderPage = this.renderPage.bind(this);
```

Comment:
You don't have to bind this method, since it's not being passed as a callback anywhere.